.github/workflows/CD.yml

```diff
@@ -54,7 +31,7 @@ jobs:
     # Split build and publish to restrict trusted publishing to just this workflow
     needs: ['pypi-build']
     name: Publish to PyPI.org
-    if: github.repository == 'payu-org/payu' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    if: always() && needs.pypi-build.result == 'success'
```

Comment:
Good question, and my answer is that I don't remember. I copied this from another workflow I wrote a while ago.

My recollection is that the behaviour of `if` conditionals in Github actions is (or at least was) a bit confusing. However, it may not be needed here. Unfortunately, I don't have time to test this. If you or someone know it's not needed, we should remove it.

Reply:
From what I can tell, I think it's not needed as `needs.pypi-build.result == 'success'` is the same as the `success()`  status check as `pypi-build` is the only previous step. But I think the code is fine as is - I just didn't know much about `always()`

Reply:
Yeah, let's remove it for clarity. I suspect it was needed at one point in the workflow I took it from, but it's not needed here. It's not really the end of the world anyway if it is needed and we try to upload a build that was unsuccessful.